File: app/src/main/java/workshop/akbolatss/tagsnews/model/DBReminderItemRepository.java
package workshop.akbolatss.tagsnews.model;


import java.util.List;
import java.util.concurrent.Callable;

import io.reactivex.Single;
import workshop.akbolatss.tagsnews.model.dao.DaoSession;
import workshop.akbolatss.tagsnews.model.dao.ReminderItem;
import workshop.akbolatss.tagsnews.model.dao.ReminderItemDao;

public class DBReminderItemRepository implements ReminderItemRepository {

    private DaoSession mDaoSession;

    public DBReminderItemRepository(DaoSession daoSession) {
        this.mDaoSession = daoSession;
    }

    @Override
    public Single<List<ReminderItem>> onLoadReminders() {
        return Single.fromCallable(new Callable<List<ReminderItem>>() {
            @Override
            public List<ReminderItem> call() throws Exception {
                ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
                return reminderItemDao.loadAll();
            }
        });
    }

    @Override
    public void onAddReminder(ReminderItem rItem) {
        ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
        reminderItemDao.insert(rItem);
    }

    @Override
    public void onRemoveReminder(ReminderItem rItem) {
        ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
        reminderItemDao.delete(rItem);
    }

    @Override
    public void onUpdateReminder(ReminderItem rItem) {
        ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
        reminderItemDao.update(rItem);
    }

    @Override
    public void onActivateReminder(ReminderItem rItem) {
        ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
        rItem.setIsActive(true);
        reminderItemDao.update(rItem);
    }

    @Override
    public void onDeactivateReminder(ReminderItem rItem) {
        ReminderItemDao reminderItemDao = mDaoSession.getReminderItemDao();
        rItem.setIsActive(false);
        reminderItemDao.update(rItem);
    }
}

File: app/src/main/java/workshop/akbolatss/tagsnews/model/DBReminderItemRepository.kt
package workshop.akbolatss.tagsnews.model


import io.reactivex.Single
import workshop.akbolatss.tagsnews.model.dao.DaoSession
import workshop.akbolatss.tagsnews.model.dao.ReminderItem

/**
 * DBReminderItemRepository is the class that contains all methods related to work with #Reminder and DataBase.
 * Repository pattern.
 * @see ReminderItem
 */
class DBReminderItemRepository(private val mDaoSession: DaoSession) : ReminderItemRepository {

    /**
     * Load list of all reminders
     */
    override fun loadReminders(): Single<List<ReminderItem>> {
        return Single.fromCallable {
            val reminderItemDao = mDaoSession.reminderItemDao
            reminderItemDao.loadAll()
        }
    }

    /**
     * Add new reminder to DB
     */
    override fun addReminder(rItem: ReminderItem) {
        val reminderItemDao = mDaoSession.reminderItemDao
        reminderItemDao.insert(rItem)
    }

    /**
     * Remove existing reminder
     */
    override fun removeReminder(rItem: ReminderItem) {
        val reminderItemDao = mDaoSession.reminderItemDao
        reminderItemDao.delete(rItem)
    }

    /**
     * Update values to existing reminder
     */
    override fun updateReminder(rItem: ReminderItem) {
        val reminderItemDao = mDaoSession.reminderItemDao
        reminderItemDao.update(rItem)
    }

    /**
     * Set active selected reminder
     */
    override fun activateReminder(rItem: ReminderItem) {
        val reminderItemDao = mDaoSession.reminderItemDao
        rItem.isActive = true
        reminderItemDao.update(rItem)
    }

    /**
     * Set inactive reminder
     */
    override fun deactivateReminder(rItem: ReminderItem) {
        val reminderItemDao = mDaoSession.reminderItemDao
        rItem.isActive = false
        reminderItemDao.update(rItem)
    }
}
